File: connection/src/main/java/cli/organization/data/geo/City.java
package cli.organization.data.geo;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * Model class for precharged cities
 *
 * @author Pablo Ahíta del Barrio
 * @version 1
 * */
public class City implements Serializable {

    /**City identifier*/
    @SerializedName("idCity")
    public int idCity;

    /**Province identifier*/
    @SerializedName("idProvince")
    public int idProvince;

    /**Region identifier*/
    @SerializedName("idRegion")
    public int idRegion;

    /**Country identifier*/
    @SerializedName("idCountry")
    public String idCountry;

    /**City name in Spanish*/
    @SerializedName("nameSpanish")
    public String nameSpanish;

    /**City name in English*/
    @SerializedName("nameEnglish")
    public String nameEnglish;

    /**City name in French*/
    @SerializedName("nameFrench")
    public String nameFrench;

    /**City name in Basque*/
    @SerializedName("nameBasque")
    public String nameBasque;

    /**City name in Catalan*/
    @SerializedName("nameCatalan")
    public String nameCatalan;

    /**City name in Dutch*/
    @SerializedName("nameDutch")
    public String nameDutch;

    /**City name in Galician*/
    @SerializedName("nameGalician")
    public String nameGalician;

    /**City name in German*/
    @SerializedName("nameGerman")
    public String nameGerman;

    /**City name in Italian*/
    @SerializedName("nameItalian")
    public String nameItalian;

    /**City name in Portuguese*/
    @SerializedName("namePortuguese")
    public String namePortuguese;

    /**
     * Class constructor
     *
     * @param idCity - City identifier
     * @param idProvince - Province identifier
     * @param idRegion - Region identifier
     * @param idCountry - Country identifier
     * @param nameSpanish - City name in Spanish
     * @param nameEnglish - City name in English
     * @param nameFrench - City name in French
     * @param nameBasque - City name in Basque
     * @param nameCatalan - City name in Catalan
     * @param nameDutch - City name in Dutch
     * @param nameGalician - City name in Galician
     * @param nameGerman - City name in German
     * @param nameItalian - City name in Italian
     * @param namePortuguese - City name in Portuguese
     * */
    public City(int idCity, int idProvince, int idRegion, String idCountry, String nameSpanish, String nameEnglish, String nameFrench, String nameBasque, String nameCatalan, String nameDutch, String nameGalician, String nameGerman, String nameItalian, String namePortuguese){
        setIdCity(idCity);
        setIdProvince(idProvince);
        setIdRegion(idRegion);
        setIdCountry(idCountry);
        setNameSpanish(nameSpanish);
        setNameEnglish(nameEnglish);
        setNameFrench(nameFrench);
        setNameBasque(nameBasque);
        setNameCatalan(nameCatalan);
        setNameDutch(nameDutch);
        setNameGalician(nameGalician);
        setNameGerman(nameGerman);
        setNameItalian(nameItalian);
        setNamePortuguese(namePortuguese);
    }

    /**
     * Method that obtains the city identifier
     *
     * @return City identifier
     * */
    public int getIdCity() {
        return idCity;
    }

    /**
     * Method that sets the new city identifier
     *
     * @param idCity - City identifier
     * */
    public void setIdCity(int idCity) {
        this.idCity = idCity;
    }

    /**
     * Method that obtains the region identifier
     *
     * @return Region identifier
     * */
    public int getIdRegion() {
        return idRegion;
    }

    /**
     * Method that sets the new region identifier
     *
     * @param idRegion - Region identifier
     * */
    public void setIdRegion(int idRegion) {
        this.idRegion = idRegion;
    }

    /**
     * Method that obtains the province identifier
     *
     * @return Province identifier
     * */
    public int getIdProvince() {
        return idProvince;
    }

    /**
     * Method that sets the new province identifier
     *
     * @param idProvince - Province identifier
     * */
    public void setIdProvince(int idProvince) {
        this.idProvince = idProvince;
    }

    /**
     * Method that obtains the country identifier
     *
     * @return Country identifier
     * */
    public String getIdCountry() {
        return idCountry;
    }

    /**
     * Method that sets the new country identifier
     *
     * @param idCountry - Country identifier
     * */
    public void setIdCountry(String idCountry) {
        this.idCountry = idCountry;
    }

    /**
     * Method that obtains the city name in Spanish
     *
     * @return City name in Spanish
     * */
    public String getNameSpanish() {
        return nameSpanish;
    }

    /**
     * Method that sets the new city name in Spanish
     *
     * @param nameSpanish - City name in Spanish
     * */
    public void setNameSpanish(String nameSpanish) {
        this.nameSpanish = nameSpanish;
    }

    /**
     * Method that obtains the city name in English
     *
     * @return City name in English
     * */
    public String getNameEnglish() {
        return nameEnglish;
    }

    /**
     * Method that sets the new city name in English
     *
     * @param nameEnglish - City name in English
     * */
    public void setNameEnglish(String nameEnglish) {
        this.nameEnglish = nameEnglish;
    }

    /**
     * Method that obtains the city name in French
     *
     * @return City name in French
     * */
    public String getNameFrench() {
        return nameFrench;
    }

    /**
     * Method that sets the new city name in French
     *
     * @param nameFrench - City name in French
     * */
    public void setNameFrench(String nameFrench) {
        this.nameFrench = nameFrench;
    }

    /**
     * Method that obtains the city name in Basque
     *
     * @return City name in Basque
     * */
    public String getNameBasque() {
        return nameBasque;
    }

    /**
     * Method that sets the new city name in Basque
     *
     * @param nameBasque - City name in Basque
     * */
    public void setNameBasque(String nameBasque) {
        this.nameBasque = nameBasque;
    }

    /**
     * Method that obtains the city name in Catalan
     *
     * @return City name in Catalan
     * */
    public String getNameCatalan() {
        return nameCatalan;
    }

    /**
     * Method that sets the new city name in Catalan
     *
     * @param nameCatalan - City name in Catalan
     * */
    public void setNameCatalan(String nameCatalan) {
        this.nameCatalan = nameCatalan;
    }

    /**
     * Method that obtains the city name in Dutch
     *
     * @return City name in Dutch
     * */
    public String getNameDutch() {
        return nameDutch;
    }

    /**
     * Method that sets the new city name in Dutch
     *
     * @param nameDutch - City name in Dutch
     * */
    public void setNameDutch(String nameDutch) {
        this.nameDutch = nameDutch;
    }

    /**
     * Method that obtains the city name in Galician
     *
     * @return City name in Galician
     * */
    public String getNameGalician() {
        return nameGalician;
    }

    /**
     * Method that sets the new city name in Galician
     *
     * @param nameGalician - City name in Galician
     * */
    public void setNameGalician(String nameGalician) {
        this.nameGalician = nameGalician;
    }

    /**
     * Method that obtains the city name in German
     *
     * @return City name in German
     * */
    public String getNameGerman() {
        return nameGerman;
    }

    /**
     * Method that sets the new city name in German
     *
     * @param nameGerman - City name in German
     * */
    public void setNameGerman(String nameGerman) {
        this.nameGerman = nameGerman;
    }

    /**
     * Method that obtains the city name in Italian
     *
     * @return City name in Italian
     * */
    public String getNameItalian() {
        return nameItalian;
    }

    /**
     * Method that sets the new city name in Italian
     *
     * @param nameItalian - City name in Italian
     * */
    public void setNameItalian(String nameItalian) {
        this.nameItalian = nameItalian;
    }

    /**
     * Method that obtains the city name in Portuguese
     *
     * @return City name in Portuguese
     * */
    public String getNamePortuguese() {
        return namePortuguese;
    }

    /**
     * Method that sets the new city name in Portuguese
     *
     * @param namePortuguese - City name in Portuguese
     * */
    public void setNamePortuguese(String namePortuguese) {
        this.namePortuguese = namePortuguese;
    }
}

File: OTEAServer/Models/City.cs
using System.Text.Json.Serialization;

namespace OTEAServer.Models
{
    /// <summary>
    /// Model class for cities
    /// Author: Pablo Ahita del Barrio
    /// Version: 1
    /// </summary>
    public class City
    {
        /// <summary>
        /// Class constructor
        /// </summary>
        /// <param name="idCity">City identifier</param>
        /// <param name="idProvince">Province identifier</param>
        /// <param name="idRegion">Region identifier</param>
        /// <param name="idCountry">Country identifier</param>
        /// <param name="nameSpanish">City name in Spanish</param>
        /// <param name="nameEnglish">City name in English</param>
        /// <param name="nameFrench">City name in French</param>
        /// <param name="nameBasque">City name in Basque</param>
        /// <param name="nameCatalan">City name in Catalan</param>
        /// <param name="nameDutch">City name in Dutch</param>
        /// <param name="nameGalician">City name in Galician</param>
        /// <param name="nameGerman">City name in German</param>
        /// <param name="nameItalian">City name in Italian</param>
        /// <param name="namePortuguese">City name in Portuguese</param>
        public City(int idCity, int idProvince, int idRegion, string idCountry, string nameSpanish, string nameEnglish, string nameFrench,string nameBasque, string nameCatalan, string nameDutch, string nameGalician, string nameGerman, string nameItalian, string namePortuguese) {
            this.idCity = idCity;
            this.idProvince = idProvince;
            this.idRegion = idRegion;
            this.idCountry = idCountry;
            this.nameSpanish = nameSpanish;
            this.nameEnglish = nameEnglish;
            this.nameFrench = nameFrench;
            this.nameBasque = nameBasque;
            this.nameCatalan = nameCatalan;
            this.nameDutch = nameDutch;
            this.nameGalician = nameGalician;
            this.nameGerman = nameGerman;
            this.nameItalian = nameItalian;
            this.namePortuguese = namePortuguese;
        }

        /// <summary>
        /// City identifier
        /// </summary>
        [JsonPropertyName("idCity")]
        public int idCity { get; set; }

        /// <summary>
        /// Province identifier
        /// </summary>
        [JsonPropertyName("idProvince")]
        public int idProvince { get; set; }

        /// <summary>
        /// Region identifier
        /// </summary>
        [JsonPropertyName("idRegion")]
        public int idRegion { get; set; }

        /// <summary>
        /// Country identifier
        /// </summary>
        [JsonPropertyName("idCountry")]
        public string idCountry { get; set; }

        /// <summary>
        /// City name in Spanish
        /// </summary>
        [JsonPropertyName("nameSpanish")]
        public string nameSpanish { get; set; }

        /// <summary>
        /// City name in English
        /// </summary>
        [JsonPropertyName("nameEnglish")]
        public string nameEnglish { get; set; }

        /// <summary>
        /// City name in French
        /// </summary>
        [JsonPropertyName("nameFrench")]
        public string nameFrench { get; set; }

        /// <summary>
        /// City name in Basque
        /// </summary>
        [JsonPropertyName("nameBasque")]
        public string nameBasque { get; set; }

        /// <summary>
        /// City name in Catalan
        /// </summary>
        [JsonPropertyName("nameCatalan")]
        public string nameCatalan { get; set; }

        /// <summary>
        /// City name in Dutch
        /// </summary>
        [JsonPropertyName("nameDutch")]
        public string nameDutch { get; set; }

        /// <summary>
        /// City name in Galician
        /// </summary>
        [JsonPropertyName("nameGalician")]
        public string nameGalician { get; set; }

        /// <summary>
        /// City name in German
        /// </summary>
        [JsonPropertyName("nameGerman")]
        public string nameGerman { get; set; }

        /// <summary>
        /// City name in Italian
        /// </summary>
        [JsonPropertyName("nameItalian")]
        public string nameItalian { get; set; }

        /// <summary>
        /// City name in Portuguese
        /// </summary>
        [JsonPropertyName("namePortuguese")]
        public string namePortuguese { get; set; }
    }
}
